### DISCUSSION TITLE
Is this the right way to run background tasks?

### DISCUSSION THREAD

**Discussion by xxf2103625** [tokio-rs/axum#1998] [Q&A] [answered]
This is a hander
```
pub async fn test(State(pool):State<deadpool_postgres::Pool>)->Html<String>{
    let p2=pool.clone();
    tokio::spawn(async move {
        let client = p2.get().await.unwrap();
        client.query("",&[]);
    });
    Html("Ok".to_owned())
}
```
I want the request to return immediately and perform certain tasks continuously in the background.
Is this the right way?

**Comment by davidpdrsn** [comment, marked as answer]
Yes.

**Reply by aurelien-clu** [reply]
Could not an async task be cancelled if we don't keep a reference to it?

I believe I read that kind of issues few years ago but I may be thinking of something else or it could have been fixed since.

(Could not find what I am talking about with a quick search)

**Reply by davidpdrsn** [reply]
No. Tokio does not cancel tasks by itself. The future you spawn has to complete or you have to abort the join handle.
